File: LeetCodeRecord/src/main/java/链表/Q0206.java
package 链表;

/*
206. 反转链表
 */
public class Q0206 {
    public ListNode reverseList(ListNode head) {
        ListNode pre=null;
        ListNode cur=head;
        ListNode nxt=null;
        while(cur!=null){
            nxt = cur.next;
            cur.next = pre;

            pre = cur;
            cur = nxt;
        }
        return pre;
    }
}

File: LeetCodeRecord/cpp/链表/Q0206.cpp
//
// Created by Administrator on 2026/9/3.
//

//反转链表

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(nullptr) {}
};

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *pre = nullptr;
        ListNode *cur = head;
        ListNode *nxt = nullptr;
        while (cur != nullptr) {
            nxt = cur->next;
            cur->next = pre;
            pre = cur;
            cur = nxt;
        }
        return pre;
    }
};

